quil-rs/src/instruction/classical.rs

```diff
@@ -5,14 +5,14 @@ use super::MemoryReference;
 #[derive(Clone, Debug, Hash, PartialEq)]
 pub struct Arithmetic {
     pub operator: ArithmeticOperator,
-    pub destination: ArithmeticOperand,
+    pub destination: MemoryReference,
```

Comment:
Nice! This change alone simplifies `Arithmetic` significantly.

Reply:
Unsurprisingly, this change was the specific motivation for doing the rest! :-)